src/ciadmin/generate/worker_pools.py

```diff
@@ -207,7 +207,11 @@ def get_azure_provider_config(
         for vmSize in vmSizes:
             loc = location.replace("-", "")
             DeploymentId = image.image_id(provider_id, "deployment_id")
-            ImageId = image.image_id(provider_id, location)
+            try:
+                version = image.image_id(provider_id, "version")
+            except Exception:
```

Comment:
It looks like it might just be `KeyError` that you want to catch here.

Reply:
Do you have any suggestions on how to catch here? 

Side note: `tc-admin generate --environment=firefoxci` ran locally without error.

Reply:
`except KeyError` should do the trick.

The main concern about catching all exceptions here is that you might end up catching some _other_ error that doesn't indicate you should take the `except` branch.